tests/test_decoder_models.py

```diff
@@ -302,3 +302,25 @@ def test_generate_adalora_no_dropout(self):
     @parameterized.expand(PeftTestConfigManager.get_grid_parameters(FULL_GRID))
     def test_passing_input_embeds_works(self, test_name, model_id, config_cls, config_kwargs):
         self._test_passing_input_embeds_works(test_name, model_id, config_cls, config_kwargs)
+
+    def test_lora_layer_replication(self):
+        model_id = "HuggingFaceM4/tiny-random-LlamaForCausalLM"
+        config_kwargs = {
+            "target_modules": ['down_proj', 'up_proj'],
+            "task_type": "CAUSAL_LM",
+            "lora_dropout": 0.0,
+            "layer_replication": [[0, 1], [0, 2], [1, 2]]
+        }
+        model = self.transformers_class.from_pretrained(model_id).to(self.torch_device)
+        config = LoraConfig(
+            base_model_name_or_path=model_id,
+            **config_kwargs,
+        )
+        assert 2 == len(model.model.layers), 'Expected 2 layers in original model.'
```

Comment:
Could you please switch the order of the check, i.e. `assert len(model.model.layers) == 2`, I find it confusing otherwise. It's like writing `if "__main__" == __name__` ;-)

Reply:
Done. Habit left over from C++. Constant on the left prevent typos like `a = 5` vs `5 = a`.